app/src/main/java/com/amaze/filemanager/filesystem/operations/extract/TarExtractOperation.java

```diff
@@ -73,13 +74,14 @@ protected void extractWithFilter(@NonNull Filter filter) throws IOException {
         inputStream = new TarArchiveInputStream(new FileInputStream(filePath));
 
         for (TarArchiveEntry entry : archiveEntries) {
-            if (!listener.isCancelled()) {
-                listener.onUpdate(entry.getName());
-                //TAR is sequential, you need to walk all the way to the file you want
-                while (entry.hashCode() != inputStream.getNextTarEntry().hashCode());
-                extractEntry(context, inputStream, entry, outputPath);
-            }
+            crashOnCancelled();
+
+            listener.onUpdate(entry.getName());
+            //TAR is sequential, you need to walk all the way to the file you want
+            while (entry.hashCode() != inputStream.getNextTarEntry().hashCode()) ;
+            extractEntry(context, inputStream, entry, outputPath);
```

Comment:
I'm seeing this piece of code repeated multiple times. Perhaps this can be abstracted into a helper method in another class (`ExtractOperationHelper`?) and be referenced from there. This will be more maintainable in the long run.

Reply:
They are not exactly the same code, and the boilerplate code necessary is too much, as is each extraction is separated and clean in its own file.